### PR TITLE
Manifestv2

### PR DESCRIPTION
**Pull request by Salamandar**
## Problem

Manifestv1 !

## Solution

Manifestv2 !

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by ericgaspar** [comment]
@Salamandar I made a branch with my [version 2](https://github.com/YunoHost-Apps/gitea_ynh/tree/version-2). I gave up at one point, so feel free to poke around (I merged your huge source list)

**Comment by Salamandar** [comment]
> @Salamandar I made a branch with my [version 2](https://github.com/YunoHost-Apps/gitea_ynh/tree/version-2). I gave up at one point, so feel free to poke around (I merged your huge source list)

Oh alright ! got it :)

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
I still have to ask gitea guys if we *really* need every minor version of gitea for database upgrades

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!tcstme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

final !

**Comment by Salamandar** [comment]
F I N N A L L Y, after 27 CI runs…

**Comment by Salamandar** [comment]
@Josue-T I have manifestv2 working o/ do you want to review ?

**Comment by Salamandar** [comment]
!bookwormtestme

File: manifest.toml
Comment (ericgaspar) [inline manifest.toml:69]: Can't we remove this? is it not the standard path?

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: conf/app.ini
Comment (ericgaspar) [reply conf/app.ini:5]: also, we could use `sudirs =[ ".ssh", "repositories", "data/avatars", "data/attachments" ]` in manifest instead of mkdir... in _common.sh

File: manifest.toml
Comment (Salamandar) [reply manifest.toml:69]: Oh yeah we can !

File: manifest.toml
Comment (Salamandar) [reply manifest.toml:79]: Eh… I like the comma at the end, that way you can append to the list without having a weird git diff of adding a comma…

File: scripts/upgrade
Comment (Josue-T) [inline scripts/upgrade:112]: Just a question: why do you do DB migration only if the version < 1.6 ?

File: manifest.toml
Comment (ericgaspar) [inline manifest.toml:55]: We need to update this to the question and choices provided : visitors, all users, admins

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: scripts/upgrade
Comment (Salamandar) [reply scripts/upgrade:112]: Hi,
I’ve exchanged with the Gitea devs about DB migration. They explained we just need to go up to 1.6.4 step by step, then we can "long jump" to the latest version ; the intermediate migration steps are now kept in the new versions.

Comment (Salamandar) [reply scripts/upgrade:112]: I’m not sure but I think it’s there : https://github.com/go-gitea/gitea/tree/main/models/migrations

File: manifest.toml
Comment (Salamandar) [reply manifest.toml]: This is fixed.

**Comment by Salamandar** [comment]
!bookwormtestme

File: scripts/upgrade
Comment (Josue-T) [reply scripts/upgrade:112]: Hello, thanks for the answer. In this case we could maybe just remove the support of upgrade from version <1.6 it could bring many simplification on upgrade. I think the 1.6 version is quite old it should be ok.

Comment (Salamandar) [reply scripts/upgrade:112]: If you’re OK with that, I’m all for it ! Maybe let’s print an error message to tell to run `yunohost app upgrade gitea -u https://github.com/yunohost-apps/gitea_ynh/commit/SOME_SHA_BEFORE_THIS_MERGE_REQUEST` for people on old versions :)

File: conf/app.ini
Comment (ericgaspar) [inline conf/app.ini:8]:
```suggestion
ENABLED = false
```

Comment (ericgaspar) [reply conf/app.ini:8]: I would rather set actions support as false. (https://forum.yunohost.org/t/has-anyone-managed-to-get-gitea-actions-working-with-yunohost/27064)

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: manifest.toml
Comment (ericgaspar) [inline manifest.toml:61]: what about adding `allow_ssh = true` as well as `allow_email = true` to system_user? (app.ini should be adapted with mail user and password)

Comment (Salamandar) [reply manifest.toml:61]: Oh ! That means I won’t have to manually add the user to app.ssh ?

Comment (ericgaspar) [reply manifest.toml:61]: https://yunohost.org/fr/packaging_apps_resources#system-user

Comment (Salamandar) [reply manifest.toml:61]: Fixed.

File: manifest.toml
Comment (ericgaspar) [inline manifest.toml:63]: and we need to adapt `app.ini` config file for mail
```
[mailer]
ENABLED = true
SMTP_ADDR = 127.0.0.1
SMTP_PORT = 25
FROM = "Gitea" <__APP__@__DOMAIN__>
USER = __APP__
PASSWD = __MAIL_PWD__
SKIP_VERIFY = true
```

Comment (Salamandar) [reply manifest.toml:78]: My bad. Fixed.

**Comment by Salamandar** [comment]
!boowormtestme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!bookwormtestme

**Comment by Josue-T** [comment]
## TODO

- [ ] Fix and understand why admin don't work any more on LDAP
- [ ] Add group support on LDAP
- [ ] Test deeply upgrade from last version (package v1)

**Comment by Salamandar** [comment]
>  Fix and understand why admin don't work any more on LDAP

What issue did you find with admin ? It works on my end.

>  Add group support on LDAP

It was not supported before right ? If so, i’d prefer to handle it in another merge request :)

>  Test deeply upgrade from last version (package v1)

It should work fine ("should"…). It worked on my server. The CI does not show particular issues, and once it’s merged on testing, we can ask on the forum for voluntaries to test it.

**Comment by Salamandar** [comment]
Note for myself : maybe call [gitea doctor convert](https://docs.gitea.com/1.21/administration/command-line#doctor-convert) on upgrade for old databases ?

**Comment by Salamandar** [comment]
@Josue-T I removed some commits of yours, here are the reasons : 
* email config: we need to talk with the local email server, via 127.0.0.1, not __DOMAIN__.
* fix manifest : we don’t need resources.install_dir.dir as it’s the default value, we already discussed that with @ericgaspar 
* cleanup app.ini : I understand the changes, I’ll just reword it and make according changes to scripts/_common.sh

**Comment by Josue-T** [comment]
>     email config: we need to talk with the local email server, via 127.0.0.1, not **DOMAIN**.

Well, I've tested all mail stack (sending AND receiving email) and it work, so please leave the config as it is.

>     fix manifest : we don’t need resources.install_dir.dir as it’s the default value, we already discussed that with @ericgaspar

By default it's stored to /var/www/$app but for me it's not a good choice to store binary file in this place, it was discussed many time, the last one was here: https://github.com/YunoHost-Apps/seafile_ynh/pull/109#discussion_r1356487263

File: conf/app.ini
Comment (Josue-T) [inline conf/app.ini:87]: Did you test fail2ban before to commit this ? Because sometime we use a specific log level to make fail2ban work correctly

**Review by Josue-T — CHANGES_REQUESTED** [review]

**Comment by ericgaspar** [comment]
> > email config: we need to talk with the local email server, via 127.0.0.1, not **DOMAIN**.
> 
> Well, I've tested all mail stack (sending AND receiving email) and it work, so please leave the config as it is.
> 
> > fix manifest : we don’t need resources.install_dir.dir as it’s the default value, we already discussed that with @ericgaspar
> 
> By default it's stored to /var/www/$app but for me it's not a good choice to store binary file in this place, it was discussed many time, the last one was here: [YunoHost-Apps/seafile_ynh#109 (comment)](https://github.com/YunoHost-Apps/seafile_ynh/pull/109#discussion_r1356487263)

If I recall correctly, the plan for future packaging is to have a specific path for apps. having `/var/www/$app` path as standard will facilitate automagic transition to this new path. (Same logic as transition from `/opt/yunohost/$app` to  `/home/yunohost.app/$apps`.

**Comment by Josue-T** [comment]
> > Fix and understand why admin don't work any more on LDAP
> 
> What issue did you find with admin ? It works on my end.

On my side no user are set as admin on new install. It's a know issue here : #76 

> > Add group support on LDAP
> 
> It was not supported before right ? If so, i’d prefer to handle it in another merge request :)

No only permissions are supported now. Link with user-group are not managed but I need to investigate what exactly is the feature of this.

> > Test deeply upgrade from last version (package v1)
> 
> It should work fine ("should"…). It worked on my server. The CI does not show particular issues, and once it’s merged on testing, we can ask on the forum for voluntaries to test it.

Well before merging to testing we MUST test this correctly. I already heard (like [this](https://github.com/YunoHost-Apps/synapse_ynh/pull/356#issuecomment-1716123188)) that it's better to merge to testing and after more people test but the issue with this strategy is that after we merge to master and there are many regression that nobody fix...

So no we must take the time to test correctly that everything work before merging this.

**Comment by ericgaspar** [comment]
> On my side no user are set as admin on new install. It's a know issue here : #76

This problem is not present in Forgejo... Maybe you could look into [Forgejo](https://github.com/YunoHost-Apps/forgejo_ynh) package as it is the same rebranded app as Gitea

**Comment by Josue-T** [comment]
Well, now I need to push some fix but I don't have any more access right. @Salamandar can you fix it ? Or does I need to create new PR with other branch ?

**Comment by Salamandar** [comment]
> Hello,
> 
> Well, now I need to push some fix but I don't have any more access right. @Salamandar can you fix it ? Or does I need to create new PR with other branch ?

Could you instead please make change requests on github ? That’ll help review and discussion :)

**Comment by Salamandar** [comment]
@Josue-T Why did you remove the [install.admin] question ? That’s probably why you don’t have admin users…

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
!testme

File: conf/app.ini
Comment (Salamandar) [reply conf/app.ini]: Indeed, that was a problem !

**Comment by Josue-T** [comment]
> > On my side no user are set as admin on new install. It's a know issue here : #76
> 
> This problem is not present in Forgejo... Maybe you could look into [Forgejo](https://github.com/YunoHost-Apps/forgejo_ynh) package as it is the same rebranded app as Gitea

Forgejo as the same issue cf: https://github.com/YunoHost-Apps/forgejo_ynh/blob/master/manifest.toml#L42

And it look like an upstream issue: https://github.com/go-gitea/gitea/issues/25985

File: doc/ADMIN.md
Comment (alexAubin) [inline doc/ADMIN.md:88]:
```suggestion
systemctl restart __APP__.service
```

File: doc/ADMIN.md
Comment (alexAubin) [inline doc/ADMIN.md:93]: Why not, but that's the same stuff for every app with a data dir ...

Anyway, nowadays `yunohost app remove` has a `--purge` option, and i think with packaging v2 there may be a modal asking admins wether or not to purge data ?

File: scripts/upgrade
Comment (alexAubin) [inline scripts/upgrade:111]:
```suggestion
ynh_systemd_action --service_name="$app" --action="stop"
```

File: manifest.toml
Comment (alexAubin) [inline manifest.toml:76]: Eeeeh but why hardcode this ? Isn't this the default value ?

File: manifest.toml
Comment (alexAubin) [inline manifest.toml:15]:
```suggestion
```

**Review by alexAubin — APPROVED** [review]

File: manifest.toml
Comment (alexAubin) [reply manifest.toml:76]: Ah nevermind this is the system_user and not the data_dir

**Comment by alexAubin** [comment]
>Would it be possible to merge this PR on testing first ?

+1, i don't know what's holding this PR right now, i made a few comments but these really are minor/trivialish stuff

**Comment by ericgaspar** [comment]
!testme

**Review by ericgaspar — APPROVED** [review]

**Comment by Josue-T** [comment]
Note I will push some fix this week 😉

**Pull request closed without merging**
